elasticsearch/client.go

```diff
@@ -106,13 +108,33 @@ func (cli *Client) AddDocument(ctx context.Context, indexName, documentType, doc
 
 }
 
+// BulkUpdate uses an HTTP post request to submit data to Elastic Search
+func (cli *Client) BulkUpdate(
+	ctx context.Context, esDestIndex string, esDestURL string, bulk []byte) ([]byte, int, error) {
+
+	uri := fmt.Sprintf("%s/%s/_bulk", esDestURL, esDestIndex)
+	jsonBody, status, err := cli.callElastic(ctx, uri, "POST", bulk)
+
+	if err != nil {
+		log.Info(ctx, "error posting bulk request", log.Data{
+			"err": err})
+		log.Error(ctx, "error posting bulk request %s", err)
+		return jsonBody, status, err
+	}
+
+	return jsonBody, status, err
```

Comment:
Question: Does this method need to return the `status` value? Could we achieve the same behaviour if we returned only `jsonBody, err`?

I don't know what context this method will be used in but presumably any non 200 response status is considered an error? If so that can be communicated to the called via an error that contains specific information about what went wrong.

I'm not trying to be picky but returning 3 params makes things more complicated, more combinations of things to check etc. I think we could simplify that by just returning 2.

Reply:
Initially thought was to capture all failed scenario including network error (unable to reach es) where status is 0 with nil err and jsonbody, possibly jsonbody can be removed. I feel using three parameters will give maximum information about any possible success/failed scenario.
Also, this implementation is going to capture 201 (Created) for creating a new index for incoming events in batches.